### PR TITLE
redis-py 3 compatibility

### PR DESCRIPTION
**Pull request by thomasst**
Fixes https://github.com/closeio/tasktiger/issues/119

* Make ZADD compatible with redis-py 3. For now we'll support both 2 & 3. In a future release we can drop 2.
* This drops Python 3.4 support so we can use the latest packages. <del>Note `click==7.0` and `redis==3.3.11` are the latest versions that run on Python 3.4, which we still support.</del>
* Also test Python 3.7
* I've had tests timeout on Python 3.8, will have to look into that separately (maybe consider upgrading the Redis version in the Circle image).

### CODE REVIEW

**Comment by AlecRosenbaum** [comment]
Python 3.4 hit end of life over a year ago and 3.5 is end of life in a few months (September 2020). Do we want to drop at least 3.4 support as we move to `tasktiger v0.11`?

**Comment by thomasst** [comment]
> Do we want to drop at least 3.4 support as we move to `tasktiger v0.11`?

Okay, I'll drop 3.4 support.

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-click==4.0
+click==7.1.2
 contextlib2==0.5.5; python_version < '3.3'
```
[the remark this answers is not in the document]
Comment (thomasst) [reply requirements.txt:2]: Python 2.7 maybe? (Which I don't think we should drop just now, but soon).

**Comment by thomasst** [comment]
> Is the only breaking change between v3 and v2 the kwargs for zadd?

https://github.com/andymccurdy/redis-py#upgrading-from-redis-py-2x-to-30

I didn't see anything else, and all the tests pass.

**Pull request merged**